File: Haskell/fibonacci.hs
import Data.Word

fibonacci :: Word64 -> Word64 -> IO()
fibonacci a b
    | a + b < b = putStrLn "Overflowed!"
    | otherwise = do 
        let c = a + b
        print c
        fibonacci b c

main :: IO()
main = do
    let a = 0 :: Word64
    print a
    let b = 1 :: Word64
    print b
    fibonacci a b
    

File: Pascal/fibonacci.pas
program Fibonacci;
var
    a : qWord;
    b : qWord;
    c : qWord;
begin
    a := 0;
    WriteLn(a);
    b := 1;
    WriteLn(b);
    while true do
    begin
        c := a + b;
        if c < a then
            begin
                WriteLn('Overflowed!');
                break;
            end;
        WriteLn(c);
        a := b;
        b := c;
    end;
end.
